springfox-schema/src/test/groovy/springfox/documentation/schema/property/field/FieldModelPropertyProviderSpec.groovy

```diff
@@ -91,4 +92,41 @@ class FieldModelPropertyProviderSpec extends Specification {
       deSerializationPropNames == ['name']
 
   }
+
+  def "Handles enum collections" () {
+
+    given:
+    def genericNamingStrategy = new DefaultGenericTypeNamingStrategy()
+    Class typeToTest = collectionEnumType()
+    def typeResolver = new TypeResolver()
+    ResolvedType resolvedType = typeResolver.resolve(typeToTest)
+    ObjectMapper mapper = objectMapperThatUsesFields()
+    def namingStrategy = new ObjectMapperBeanPropertyNamingStrategy()
+    namingStrategy.onApplicationEvent(new ObjectMapperConfigured(this, mapper))
+    def propertyProvider = new FieldModelPropertyProvider(new FieldProvider(typeResolver),
+            namingStrategy, defaultSchemaPlugins(), Mock(TypeNameExtractor))
+    propertyProvider.objectMapper = mapper
+
+    def serializationProps =
+            propertyProvider.propertiesFor(resolvedType, returnValue(resolvedType, DocumentationType.SWAGGER_12, alternateTypeProvider(), genericNamingStrategy))
+
+    def deSerializationProps =
+            propertyProvider.propertiesFor(resolvedType, inputParam(resolvedType, DocumentationType.SWAGGER_12, alternateTypeProvider(), genericNamingStrategy))
+
+
+    expect:
+    def assertModelProperties = { props ->
+      assert props.size() == 1
+      assert props.first().name == 'exampleEnums'
+      assert props.first().type.erasedType == List
+      assert props.first().type.typeParameters.size() == 1
+      assert props.first().type.typeParameters.first().erasedType == ExampleEnum
+      assert props.first().allowableValues.values == ['ONE', 'TWO']
+      true
+    }
+
+    assertModelProperties(serializationProps)
+    assertModelProperties(deSerializationProps)
+  }
+
```

Comment:
@dilipkrish I updated the test case a bit, I'm not if there is a better way to accomplish this.